File: lab2/Specialties.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity.Spatial

Partial Public Class Specialties
    Public Sub New()
        Students = New HashSet(Of Students)()
    End Sub

    <Key>
    Public Property ID_Specialties As Integer

    Public Property Title As String

    Public Property Description As String

    Public Overridable Property Students As ICollection(Of Students)
End Class

File: WebApplication1/Models/Specialties.cs
namespace WebApplication1
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    public partial class Specialties
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public Specialties()
        {
            Students = new HashSet<Students>();
        }

        [Key]
        public int ID_Specialties { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Students> Students { get; set; }
    }
}
